config/v1/types_infrastructure.go

```diff
@@ -1699,7 +1699,8 @@ type PowerVSServiceEndpoint struct {
 	// Power Cloud - https://cloud.ibm.com/apidocs/power-cloud
 	//
 	// +required
-	// +kubebuilder:validation:Enum=CIS;COS;COSConfig;DNSServices;GlobalCatalog;GlobalSearch;GlobalTagging;HyperProtect;IAM;KeyProtect;Power;ResourceController;ResourceManager;VPC
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern=`^[a-z0-9-]+$`
```

Comment:
If you include capital letters here, then the installer could at least validate for the correct pattern for the enum and we could get closer to having this feature working

You could also split the service endpoint and have a status version with different validation, and only revert the validation of the status version of the struct